l3kernel/l3cctab.dtx

```diff
@@ -797,12 +797,16 @@
 %
 % \begin{macro}{\cctab_const:Nn,\cctab_const:cn}
 %  Creates a new \meta{cctab~var} then sets it with the current and
-%  user-supplied codes.
+%  user-supplied codes. The \cs{debug_suspend:} and \cs{debug_resume:}
+%  functions prevent \cs{cctab_new:N} and \cs{cctab_gset:Nn} from
+%  raising errors on global assignment to constant.
 %    \begin{macrocode}
 \cs_new_protected:Npn \cctab_const:Nn #1#2
   {
+    \debug_suspend:
     \cctab_new:N #1
     \cctab_gset:Nn #1 {#2}
+    \debug_resume:
   }
```

Comment:
This may suspend too much.

How about
- adding `\__kernel_intarray_new:Nn` which is let to `\intarray_new:Nn` but not patched by l3debug and
- defining `\cctab_const:Nn` with actual, two-versioned implementations of `\cctab_new:N` and `\cctab_gset:Nn`?